python_modules/dagster/dagster/_core/storage/asset_value_loader.py

```diff
@@ -98,7 +99,10 @@ def load_asset_value(
         )
         io_manager_key = assets_def.get_io_manager_key_for_asset_key(asset_key)
         io_manager_def = resource_defs[io_manager_key]
-        required_resource_keys = io_manager_def.required_resource_keys | {io_manager_key}
+        get_transitive_required_resource_keys(io_manager_def.required_resource_keys, resource_defs)
```

Comment:
Is this line on purpose?

Reply:
yes - this is what enables the transitive resource dependencies. They didn't work before this.

Reply:
nope